language/stdlib/modules/libra.move

```diff
@@ -288,13 +288,15 @@ module Libra {
     // Permanently remove the coins held in the passed-in preburn resource
     // and update the market cap accordingly. If there are multiple preburn
     // requests in progress, this will remove the oldest one.
-    // Can only be invoked by the holder of the `BurnCapability`. Fails if
-    // the `preburn` resource has no pending burn requests.
+    // Can only be invoked by the holder of the `BurnCapability`. Does
+    // nothing if the `preburn` resource has no pending burn requests.
     public fun burn_with_resource_cap<Token>(
         preburn: &mut Preburn<Token>,
         preburn_address: address,
         _capability: &BurnCapability<Token>
     ) acquires CurrencyInfo {
+        // If there aren't any preburn requests don't try to burn any
+        if (Vector::is_empty(&preburn.requests)) return;
```

Comment:
Would it be possible for this to be an assert rather than a silent return? My sense is that failing loudly in this case rather than succeeding + doing nothing is easier to debug (and easier to specify for the prover as well).

Reply:
It's a good question and I was going back and forth. If we don't have this we'll need to have some kind of "num_preburn_requests" function to return the number of preburn requests in a preburn resource (since if no txn fees in a given currency have been collected the burn will abort)